### PR TITLE
feat(unlock-app, unlock-js, locksmih) - strict check lock type from attributes

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description
We need a better way to check if a lock is an event/certification ecc...
Added an endpoint that returns the lock types based on the attributes properties.

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: locksmith/src/operations/wedlocksOperations.ts [hunk 1]
```diff
@@ -192,8 +195,10 @@ const getAttachments = async ({
     attachments.push({ path: dataURI })
   }
 
+  const { isEvent } = types ?? {}
+
   // Add ICS attachment when event is present
-  if (event) {
+  if (isEvent && event) {
```
Comment (kalidiagne) [inline locksmith/src/operations/wedlocksOperations.ts:201]: add `event` attachment only when lock is eventType

File: unlock-app/src/components/content/event/EventDetails.tsx [hunk 2]
```diff
@@ -74,7 +79,7 @@ export const EventDetails = ({ lockAddress, network }: EventDetailsProps) => {
     )
   }
 
-  if (!metadata?.attributes) {
+  if (!metadata && !types?.isEvent) {
```
Comment (kalidiagne) [inline unlock-app/src/components/content/event/EventDetails.tsx:82]: show details only when lock `isEvent` and have metadata

File: locksmith/src/controllers/v2/metadataController.ts [hunk 3]
```diff
@@ -246,3 +246,17 @@ export const updateUsersMetadata: RequestHandler = async (
     error,
   })
 }
+
+export const getLockTypeByMetadata: RequestHandler = async (
```
Comment (julien51) [inline locksmith/src/controllers/v2/metadataController.ts:250]: Let's do this check on the front-end

**Review by julien51 — CHANGES_REQUESTED** [review]
It's a good idea but there is no reason to do it on locksmith. Can we do it on the front-end please? This will save resources and reduce our reliance on the backend.

**Comment by kalidiagne** [comment]
> It's a good idea but there is no reason to do it on locksmith. Can we do it on the front-end please? This will save resources and reduce our reliance on the backend.

Actually, we need to do the check also from locksmith, to check for the event type before adding the attachments.

File: unlock-app/src/components/content/event/utils.tsx [hunk 4]
```diff
@@ -24,3 +25,26 @@ export const getEventEndDate = (ticket: any): Date | null => {
...
   return null
 }
+
+export interface LockTypes {
+  isEvent: boolean
+  isCertification: boolean
+  isStamp: boolean
+}
+
+export const getLockTypeByMetadata = (
```
Comment (kalidiagne) [inline unlock-app/src/components/content/event/utils.tsx:35]: check on frontend with metadata without endpoint call

File: locksmith/src/operations/metadataOperations.ts [hunk 5]
```diff
@@ -272,3 +272,32 @@ export const getLockMetadata = async ({
...
   return lockData?.data
 }
+
+export interface LockTypes {
+  isEvent: boolean
+  isCertification: boolean
+  isStamp: boolean
+}
+
+export const getLockTypeByMetadata = async ({
```
Comment (kalidiagne) [inline locksmith/src/operations/metadataOperations.ts:282]: helper to check metadata attributes on locksmith

**Comment by kalidiagne** [comment]
> It's a good idea but there is no reason to do it on locksmith. Can we do it on the front-end please? This will save resources and reduce our reliance on the backend.

Update and remove the endpoint

File: locksmith/src/operations/metadataOperations.ts [hunk 6]
```diff
@@ -272,3 +272,32 @@ export const getLockMetadata = async ({
...
+    return attributes.some((attribute) =>
+      attribute?.trait_type?.startsWith(name)
+    )
+  }
+
+  return {
+    isEvent: hasAttribute('event'),
+    isCertification: hasAttribute('certification'),
+    isStamp: hasAttribute('stamp'),
+  }
```
Comment (searchableguy) [inline locksmith/src/operations/metadataOperations.ts:302]: Wanna move this to `core` library?

**Comment by julien51** [comment]
> Actually, we need to do the check also from locksmith, to check for the event type before adding the attachments.

Ok but then do not make an API for this. Just a re-usable library.

File: packages/core/src/utils/index.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import { z } from 'zod'
+
+const LockTypeSchema = z.object({
+  isEvent: z.boolean().default(false),
+  isCertification: z.boolean().default(false),
+  isStamp: z.boolean().default(false),
+})
+
+export const getLockTypeByMetadata = (metadata: any): LockType => {
```
Comment (julien51) [inline packages/core/src/utils/index.ts:9]: Please add tests for this.

File: locksmith/src/operations/wedlocksOperations.ts (hunk 1 above)
Comment (julien51) [reply locksmith/src/operations/wedlocksOperations.ts:202]: Great!

**Review by julien51 — APPROVED** [review]
Please add tests to the `core` library function you're adding.
This is pretty central.

**Comment by kalidiagne** [comment]
Test in place!

File: packages/core/package.json [hunk 8]
```diff
@@ -4,13 +4,18 @@
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
   "scripts": {
-    "build": "tsc -p ."
+    "build": "tsc -p .",
+    "test": "UNLOCK_ENV=test vitest run --coverage --environment=node"
```
Comment (julien51) [inline packages/core/package.json:8]: Please also add `ci` like we do for other packages (and make sure it runs the linter as well!)

**Comment by kalidiagne** [comment]
includes also https://github.com/unlock-protocol/unlock/issues/11594

**Comment by julien51** [comment]
I don't think tests are running for the core package

**Pull request merged**